weaver/cli.py

```diff
@@ -0,0 +1,747 @@
+        :returns: results of the operation.
+        """
+        result = self._parse_deploy_body(body, process_id)
+        if not result.success:
+            return result
+        headers = copy.deepcopy(self._headers)
+        headers.update(self._parse_auth_token(token, username, password))
+        data = result.body
+        try:
+            p_id = data.get("processDescription", {}).get("process", {}).get("id", process_id)
+            info = {"id": p_id}  # minimum requirement for process offering validation
+            if isinstance(cwl, str) or isinstance(wps, str):
+                LOGGER.debug("Override loaded CWL into provided/loaded body for process: [%s]", process_id)
+                proc = get_process_definition(info, reference=cwl or wps, headers=headers)  # validate
+                data["executionUnit"] = [{"unit": proc["package"]}]
+            elif isinstance(cwl, dict):
+                LOGGER.debug("Override provided CWL into provided/loaded body for process: [%s]", process_id)
+                get_process_definition(info, package=cwl, headers=headers)  # validate
+                data["executionUnit"] = [{"unit": cwl}]
+        except PackageRegistrationError as exc:
+            message = f"Failed resolution of package definition: [{exc!s}]"
+            return OperationResult(False, message, cwl)
+        base = self._get_url(url)
+        if undeploy:
+            LOGGER.debug("Performing requested undeploy of process: [%s]", process_id)
+            result = self.undeploy(process_id=process_id, url=base)
+            if result.code not in [200, 404]:
+                return OperationResult(False, "Failed requested undeployment prior deployment.",
+                                       body=result.body, text=result.text, code=result.code, headers=result.headers)
+        path = f"{base}/processes"
+        resp = request_extra("POST", path, json=data, headers=headers, settings=self._settings)
+        return self._parse_result(resp)
+
+    def undeploy(self, process_id, url=None):
+        # type: (str, Optional[str]) -> OperationResult
+        """
+        Undeploy an existing :term:`Process`.
+
+        :param process_id: Identifier of the process to undeploy.
+        :param url: Instance URL if not already provided during client creation.
+        """
+        base = self._get_url(url)
+        path = f"{base}/processes/{process_id}"
+        resp = request_extra("DELETE", path, headers=self._headers, settings=self._settings)
+        return self._parse_result(resp)
+
+    def capabilities(self, url=None):
+        # type: (Optional[str]) -> OperationResult
+        """
+        List all available :term:`Process` on the instance.
+
+        :param url: Instance URL if not already provided during client creation.
+        """
+        base = self._get_url(url)
+        path = f"{base}/processes"
+        query = {"detail": False}  # not supported by non-Weaver, but save the work if possible
```

Comment:
Sadly, the check must still be done each time for the CLI since it won't preserve the state. 

I'm more concerned about the large size here. 
If there are hundreds of processes, this slows down quickly considering how many details are provided per process. 
The server could even force paging in such cases, (as offered by #385), though Weaver doesn't force it now.

The operation could retry the operation without query params if it fails with bad request.
Otherwise, I can also push it toward OGC and hopefully it becomes a recommendation that implementers migth pick up.

Reply:
Given that it's Weaver's CLI, I don't mind that the CLI is Weaver oriented. And since the API is still a moving target we don't have to be fully compliant at this point either.